static/js/app.js

```diff
@@ -3,106 +3,158 @@
+    //build the annotation
+    let content = extractSelectedContent(range);
+    let notation = new Annotation(
+      range,
+      content,
+      currentLabel
+    );
+
+    //then add this annotation to the current document
+    currentDocument.annotations.push(notation);
+
+    renderTextareaHighlights();
+  }
 }
 
 //returns the starting and ending position of the currently selected text
 getRangeOfSelectedText = function () {
-    let start = textArea.selectionStart;
-    let end = textArea.selectionEnd;
-    return {
-        "startPosition": start,
-        "endPosition": end
-    }
+  let start = textArea.selectionStart;
+  let end = textArea.selectionEnd;
+  return {
+    "startPosition": start,
+    "endPosition": end
+  }
 };
 
-function selectedInputRangeIsValid(range){
-  console.log(range.startPosition, range.endPosition);
-    return range.startPosition !== range.endPosition;
+function selectedInputRangeIsValid(range) {
+  console.log(range.startPosition, range.endPosition);              //remove when done
+  return range.startPosition !== range.endPosition;
 }
 
-function extractSelectedContent(range){
-    return textArea.value.substring(range.startPosition, range.endPosition);
+function extractSelectedContent(range) {
+  return textArea.value.substring(range.startPosition, range.endPosition);
 }
 
 //Actually draws the highlights on the textarea.
 renderTextareaHighlights = function () {
-    //array to hold everything that needs to be highlighted in doc
-    let highlights = [];
+  //array to hold everything that needs to be highlighted in doc
+  let highlights = [];
 
-    //loop through the stored annotations and append to the array as a dictionary
-    currentDocument.annotations.forEach(function(annotation) {
-        let single_highlight = {};
+  //loop through the stored annotations and append to the array as a dictionary
+  currentDocument.annotations.forEach(function (annotation) {
+    let single_highlight = {};
 
-        single_highlight.highlight = [annotation.range.startPosition, annotation.range.endPosition];
-        single_highlight.className = annotation.highlighter_color;
+    single_highlight.highlight = [annotation.range.startPosition, annotation.range.endPosition];
+    single_highlight.className = "label_" + annotation.label;
 
-        highlights.push(single_highlight);
-    });
+    highlights.push(single_highlight);
+  });
 
-    //then highlight based on that array
-    $('textarea').highlightWithinTextarea({
-        highlight: highlights
-    });
+  //then highlight based on that array
+  $('textarea').highlightWithinTextarea({
+    highlight: highlights
+  });
 };
 
 //change the document's label context
-$('input[type=radio]').change(function () {
-    currentLabel = this.value;
+$('.label').on('click', function () {
+  console.log(this.getAttribute('value'));                        //remove when done
+  console.log($(this).css('background-color'));                   //remove when done
```

Comment:
Again we want to keep these but enrich

Reply:
updated console logging to be more descriptive x2